test-integration/test_integration/test_redis_queue.py

```diff
@@ -2084,6 +2084,80 @@ def capture_response(request):
         )
 
 
+def test_queue_worker_webhook_events(
```

Comment:
Are you sure we want to add an integration test for this? If it's not possible to unit test it I get it, but it seems like maybe a refactoring like the one I suggested above might allow us to gain confidence that it works without having to add another (slow) integration test.

Reply:
Going to leave the integration test in for now, awaiting the large refactor described above.